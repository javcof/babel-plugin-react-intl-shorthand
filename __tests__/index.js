import path from 'path'
import { transform } from 'babel-core'
import plugin from '../src'

const filename = path.resolve(__dirname, 'index.js')

it('works', () => {
  const example = `
    import { defineMessages } from 'react-intl'

    defineMessages({
        hello: {
          id: 'hello',
          defaultMessage: 'hello world',
        },
    })
  `
  const options = {
    plugins: [plugin],
    filename,
  }

  const { code } = transform(example, options);
  expect(code).toMatchSnapshot();
})
