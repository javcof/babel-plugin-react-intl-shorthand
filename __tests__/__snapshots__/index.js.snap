// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"'use strict';

var _reactIntl = require('react-intl');

(0, _reactIntl.defineMessages)({
  hello: {
    id: 'hello',
    defaultMessage: 'hello world'
  }
});"
`;
